brave/brave-core
# Replace Wallet P3A onboarding metric, add new metric for new user balance

Resolves https://github.com/brave/brave-browser/issues/30292



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

For onboarding metric:

0. Start out with fresh profile for each test
1. Ensure the metric does not exist in local state
2. Test each answer by stopping at a particular step in the onboarding process and waiting 120 seconds. Verify that the metric still does not exist before the 120 second deadline. After the deadline, verify that the metric exists with the correct answer.
3. Repeat process for each onboarding step

Additional test:

0. Start out with fresh profile
1. Pause in the middle of the onboarding process for 120 seconds. Verify that the metric has been recorded.
2. Advance to the next step in the onboarding, verify that the metric is updated immediately.

For new user balance metric:

0. Start with fresh profile, use `--p3a-count-wallet-test-networks`
1. Setup wallet, ensure metric does not exist.
2. Send testnet ETH to wallet. Reload wallet page and view the balance.
3. Ensure metric exists with value `1`

Ensure both metrics are removed once they are sent and a month has passed.
